worker/src/RTC/Codecs/VP9.cpp

```diff
@@ -281,9 +281,19 @@ namespace RTC
 				}
 			}
 
-			// Filter spatial layers higher than current one (unless old packet).
-			if (packetSpatialLayer > tmpSpatialLayer && !isOldPacket)
-				return false;
+			// Unless old packet filter spatial layers that are either
+			// * higher than current one
+			// * different than the current one when KSVC is enabled and this is not a keyframe
+			// (interframe p bit = 1)
+			if (!isOldPacket)
+			{
+				if (
+				  packetSpatialLayer > tmpSpatialLayer ||
+				  (context->IsKSvc() && this->payloadDescriptor->p && packetSpatialLayer != tmpSpatialLayer))
```

Comment:
Please merge both conditions into one. No reason for having a if inside another if.